packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkEncodingGeneric.ts

```diff
@@ -143,27 +145,27 @@ export function handleShapesAndIdentifiers<TEncodedShape>(
  * Comparison by content would be difficult due to shape containing references to other shapes.
  *
  * @privateRemarks
- * Unlike with identifiers, conversion from the initial form (this class / IdentifierToken) is done by the `encodeShape` method, not by general purpose logic in `handleShapesAndIdentifiers`.
- * For `handleShapesAndIdentifiers` to do the conversion without help from `encodeShape`,
+ * Unlike with identifiers, conversion from the initial form (this class / IdentifierToken) is done by the `encodeShape` method, not by general purpose logic in `encodeShapesAndIdentifiers`.
+ * For `encodeShapesAndIdentifiers` to do the conversion without help from `encodeShape`,
  * instances of this Shape class would have to either be or output an object that is identical to the `TEncodedShape` format except with all shape references as object references instead of indexes.
  * Those objects would have to be deeply traversed looking for shape objects to replace with reference indexes.
  * This is possible, but making it type safe would involve generating derived types from the `TEncodedShape` deeply replacing any shape references, as well as requiring deep traversal of all objects in the encoded output.
  * Such an approach seemed less maintainable and readable than the design taken here which avoids the need for those derived types.
  */
 export abstract class Shape<TEncodedShape> {
 	/**
-	 * Count this shape's contents.
+	 * Discover referenced shapes and call the provided callback for each discovered shape. Also, count this shape's contents.
```

Comment:
I don't think describing this as two separate operations this way makes sense. This tracks what shapes and what identifiers are referenced, and how many times. With these renames and rephrasing, I think its less clear that the number of times a shape is referenced matters.

Maybe:

```suggestion
	 * Enumerate all contents of this shape that can get replaced with references (for better compression/deduplication) during encoding.
	 * This currently includes shapes and identifiers.
```